samza-core/src/main/java/org/apache/samza/operators/spec/OperatorSpecs.java

```diff
@@ -100,77 +100,75 @@ public void init(Config config, TaskContext context) {
       public void init(Config config, TaskContext context) {
         filterFn.init(config, context);
       }
-    }, output, OperatorSpec.OpCode.FILTER, graph.getNextOpId());
+    }, nextStream, OperatorSpec.OpCode.FILTER, opId);
   }
 
   /**
    * Creates a {@link StreamOperatorSpec}.
    *
    * @param transformFn  the transformation function
-   * @param graph  the {@link StreamGraphImpl} object
-   * @param output  the output {@link MessageStreamImpl} object
+   * @param nextStream  the output {@link MessageStreamImpl} to send messages to
+   * @param opId  the unique ID of the operator
    * @param <M>  type of input message
    * @param <OM>  type of output message
    * @return  the {@link StreamOperatorSpec}
    */
   public static <M, OM> StreamOperatorSpec<M, OM> createStreamOperatorSpec(
-      FlatMapFunction<M, OM> transformFn, StreamGraphImpl graph, MessageStreamImpl<OM> output) {
-    return new StreamOperatorSpec<>(transformFn, output, OperatorSpec.OpCode.FLAT_MAP, graph.getNextOpId());
+      FlatMapFunction<M, OM> transformFn, MessageStreamImpl<OM> nextStream, int opId) {
+    return new StreamOperatorSpec<>(transformFn, nextStream, OperatorSpec.OpCode.FLAT_MAP, opId);
   }
 
   /**
-   * Creates a {@link SinkOperatorSpec}.
+   * Creates an {@link OutputOperatorSpec} for the sink operator.
    *
-   * @param sinkFn  the sink function
+   * @param sinkFn  the sink function provided by the user
+   * @param opId  the unique ID of the operator
    * @param <M>  type of input message
-   * @param graph  the {@link StreamGraphImpl} object
-   * @return  the {@link SinkOperatorSpec}
+   * @return  the {@link OutputOperatorSpec} for the sink operator
```

Comment:
It seems that you have renamed the SinkOperatorSpec to OutputOperatorSpec w/o changing the opCode correspondingly. Any particular reason? I thought that InputOperatorSpec/OutputOperatorSpec is not in the scope of this change?

Following the line of thought on sink() or sendTo() methods, here are the few things that we had talked about in another PR #51 
- if sink() is only send to non-stream terminals, maybe we should define SinkFunction as SinkFunction(M message, TaskCoordinator coordinator), removing the MessageCollector parameter there?
- For sendTo(), if we implemented SAMZA-1101, it would be translated into an OutputOperatorSpec, which only contains StreamSpec, keyExtractor, msgExtractor, instead of an actual SinkFunction?

If the answers to the above questions are not intended in this PR, I would prefer to move the changes to address SAMZA-1101 to a separate one